### ISSUE TITLE
Xcode_16.app does not contain `Contents/Developer`

### ISSUE THREAD

**Issue by jeromelaban** [actions/runner-images#10918]
### Description

Running:
```
/bin/bash -c "sudo xcode-select -s /Applications/Xcode_16.app/Contents/Developer"
```

fails with a missing folder, 16.1 as well.

### Platforms affected

- [X] Azure DevOps
- [ ] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] Ubuntu 24.04
- [ ] macOS 12
- [ ] macOS 13
- [ ] macOS 13 Arm64
- [X] macOS 14
- [ ] macOS 14 Arm64
- [ ] macOS 15
- [ ] macOS 15 Arm64
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

Fails with image version: 20241106.300
Worked with image version: 20241022.254

### Is it regression?

Yes

### Expected behavior

xcode-select can select the xcode version.

### Actual behavior

Selection fails:

```
Xcode Root to /Applications/Xcode_16.app
xcode-select: error: invalid developer directory '/Applications/Xcode_16.app/Contents/Developer
```

### Repro steps

```
/bin/bash -c "sudo xcode-select -s /Applications/Xcode_16.app/Contents/Developer"
```

**Comment by jeromelaban** [comment]
This is a major blocker, any workaround would be appreciated. Thanks!

**Comment by susmitamane** [comment]
Hi @jeromelaban We will look into the issue and keep you posted with updates.

**Comment by jeromelaban** [comment]
Thanks, this makes sense.

I wish it was possible to get GitHub notifications that track issues/discussions with a specific tag (e.g. `announcement`), and not everything else, this would make this kind of changes less of a surprise...

**Comment by jeromelaban** [comment]
As always changing versions of macOS is causing massive breaking changes all around.

Things like:
```
AVPIdentity: AppleVirtualPlatformHostKey.mm:234: Assert: platformExpert,  value: 0
AVPIdentity: AppleVirtualPlatformHostKey.mm:234: Failed to get AppleVirtualPlatformARMPE service.
```

when starting simulators don't happen with macOS 14.

**Comment by jeromelaban** [comment]
@erik-bershel for what I can see on the main page of this repo, the previous images are being restored (it's an october image) to the agents, is that correct?

**Comment by erik-bershel** [comment]
> @erik-bershel for what I can see on the main page of this repo, the previous images are being restored (it's an october image) to the agents, is that correct? Thanks!

Nope. Rollout is in progress - it's time consuming process unfortunately. [Release page](https://github.com/actions/runner-images/releases/tag/macos-14%2F20241106.300). [PR](https://github.com/actions/runner-images/pull/10916) to be merged when all the environments updated.

> I wish it was possible to get GitHub notifications that track issues/discussions with a specific tag (e.g. announcement), and not everything else, this would make this kind of changes less of a surprise...

Historically, there has been a rather contradictory situation. Many users want to receive this kind of notifications, but at the same time do not want to receive all updates at all, but only those that concern their projects. Which is not feasible at the moment, to my regret. 😞 
You can subscribe to track new releases in this repository, although this will not give much time to react. 😭 
Perhaps we will move to the practice of notifying users about upcoming changes known to us in run notifications - this mechanism is under discussion, but no promises, since it also has its drawbacks.

> Things like:
>
>AVPIdentity: AppleVirtualPlatformHostKey.mm:234: Assert: platformExpert,  value: 0
>AVPIdentity: AppleVirtualPlatformHostKey.mm:234: Failed to get AppleVirtualPlatformARMPE service.
>when starting simulators don't happen with macOS 14.
>
>Edit: The message is unrelated to my build errors, which is caused iOS 17-X emulators not being present in that build.
>Edit 2: This message is actually causing lots of trouble as it ends up being added to the simulators lists command output causing all sorts of parse trouble.

It would be great if you could open a separate issue on this with more details like logs to work on and test. Ideally, you could provide the minimum steps to reproduce.

**Comment by jeromelaban** [comment]
Thanks for the detailed response.

At this time, as we can't upgrade to macos-15, nor rollback to previous macos-14 (.NET 9 prevents it) we need to disable all of our testing suited that use macOS, and that is a large problem. I'll open an issue for the assertion message.
